src/twisted/conch/test/test_session.py

```diff
@@ -705,10 +722,29 @@ def test_requestPty(self):
         self.assertSessionIsStubSession()
         self.assertRequestRaisedRuntimeError()
         # 'good' terminal type succeeds
-        self.assertTrue(self.session.requestReceived(b'pty_req',
+        self.assertTrue(self.session.requestReceived(
+            b'pty_req',
             session.packRequest_pty_req(b'good', (1, 2, 3, 4), b'')))
-        self.assertEqual(self.session.session.ptyRequest,
-                (b'good', (1, 2, 3, 4), []))
+        self.assertEqual(
+            self.session.session.ptyRequest, (b'good', (1, 2, 3, 4), []))
+
+
+    def test_setEnv(self):
+        """
+        When a client requests passing an environment variable, the
+        SSHSession object should make the request by getting an ISession
+        adapter for the avatar, then calling setEnv with the environment
+        variable name and value.
```

Comment:
This test doesn't seem to cover the `ValueError` code path that silently ignores a variable. We need that coverage.

Reply:
Done.